core/src/main/scala/api/sfs.scala

```diff
@@ -25,15 +30,19 @@ trait Filesystem {
    *  things too eagerly, nor caching data for too long.
    */
   def resolve(path: Path): Key
-  def metadata(key: Key): Metadata
-  def lookup(key: Key): Data
+  def metadata(key: Key): M[Metadata]
+  def lookup(key: Key): M[Data]
```

Comment:
Yeah, I thought about that as well. I just wondered if that would needlessly complicate things further.

Do we make it too complicated by allow effects to influence the construction of metadata, or do we make it too complicated by forcing effects into keys. I think all cards are on the table, I'll think on it.